include/openssl/asn1.h

```diff
@@ -678,6 +678,42 @@ void *ASN1_dup(i2d_of_void *i2d, d2i_of_void *d2i, const void *x);
                      CHECKED_PTR_OF(const type, x)))
 
 void *ASN1_item_dup(const ASN1_ITEM *it, const void *x);
+int ASN1_item_digest(const ASN1_ITEM *it, const EVP_MD *type, void *data,
+                     unsigned char *md, unsigned int *len);
+
+DEPRECATEDIN_3_0(int ASN1_verify(i2d_of_void *i2d, X509_ALGOR *algor1,
+                                 ASN1_BIT_STRING *signature, char *data,
+                                 EVP_PKEY *pkey))
+
+DEPRECATEDIN_3_0(int ASN1_digest(i2d_of_void *i2d, const EVP_MD *type,
+                                 char *data,
+                                 unsigned char *md, unsigned int *len))
+
+DEPRECATEDIN_3_0(int ASN1_sign(i2d_of_void *i2d, X509_ALGOR *algor1,
+                               X509_ALGOR *algor2, ASN1_BIT_STRING *signature,
+                               char *data, EVP_PKEY *pkey, const EVP_MD *type))
+
+int ASN1_item_sign(const ASN1_ITEM *it, X509_ALGOR *algor1, X509_ALGOR *algor2,
+                   ASN1_BIT_STRING *signature, const void *data,
+                   EVP_PKEY *pkey, const EVP_MD *md);
+int ASN1_item_sign_ex(const ASN1_ITEM *it, X509_ALGOR *algor1,
```

Comment:
Anywhere where we have a libctx added use 'with_libctx' e.g. 
ASN1_item_sign_with_libctx() and
ASN1_item_verify_with_libctx()

Also can we really move these form another public header... That feels like a braking change?

Reply:
> Anywhere where we have a libctx added use 'with_libctx' e.g.
> ASN1_item_sign_with_libctx() and
> ASN1_item_verify_with_libctx()

Done this rename.

Reply:
> Also can we really move these from another public header... That feels like a breaking change?

Well, could be, so I've reverted this cleanup.